### ISSUE TITLE
Use better verbs on buttons in edit entry/group widgets

### ISSUE THREAD

**Issue by newhinton** [keepassxreboot/keepassxc#6810]
## Summary
The edit-view currently has three buttons:
Ok, Apply and Close

One of them is redundant.

If the entry is changed, Ok serves the same function as pressing Apply&Close.
If it is not changed, the close button is the only button serving a function.

I propose to remove the OK-button, and rename the apply-button to save. 
This makes the function of both buttons clear, and removes ambiguity from the ui.
If the entry is not changed, the save button should be greyed out.
To preserve functionality, the Ctrl+S-shortcut should be introduced, and the save-button could get a dropdown, where the user can switch to 'Save&Exit'. This selection can be preserved.

**Comment by droidmonkey** [comment]
Save has a very specific meaning in our app, the database is written from memory to disk. It would not be correct to use that term in the edit dialog.  OK could be relabeled "Apply & Close".

We also have not received any feedback like you mention in a few usability studies. For better or worse we have conditioned computer users to associate OK with "In done, accept my changes"

**Comment by newhinton** [comment]
I don't think the argument 'save has a specific' meaning holds true. The difference between saving the database and saving an entry is a technical one, not really percievable by the user. Also, since the default is to save changes to disk after each change, makes the original save a rather rare encounter. In the case with autosave, it is actually exactly what happens: pressing apply saves to disk.

I think the buttons are mislabeled, and there is still the "problem" that there is a second button which has no meaning at all, because Cancel and OK behave the same. 

(Mislabeled because Cancel is only the correct term when there have been changes, it should be close otherwise. Maybe it would be an idea to only show "close" on unmodified entries, and "ok, apply, cancel" on modified ones. That would remove the ambiguity aswell, to the tradeoff that there is a dynamic element in the ui)

Please dont take this argument too serious, i really like the application in every aspect and i am currently nitpicking on the small stuff ;) (and willing to create PR's for it)

**Comment by droidmonkey** [comment]
Cancel is also incorrect, it should be discard or close as you mentioned. The verbs that make the most sense are: "Accept", "Apply", and "Discard/Close". Accept being apply and close "I'm done" versus just Apply "I'm still working but wanted to save state".

**Comment by JohnLGalt** [comment]
I like the idea of using `Accept` `Apply` and `Discard` in this situation.

**Comment by newhinton** [comment]
I tried to add ctrl-s as a shortcut. However, while it works and saves the entry, the buttons on the bottom are not updated until i save a second time. Why does it not reflect the state on the first save?

```
QShortcut *shortcut = new QShortcut(QKeySequence::Save, m_ui->buttonBox);
connect(shortcut, SIGNAL(activated()), SIGNAL(apply()));
```

**Comment by droidmonkey** [comment]
Ctrl + S is bound to saving the database. That is not a good shortcut to use for anything else since that shortcut is window based.

**Comment by newhinton** [comment]
I disagree. The default of keepass is to save the database instantly when an entry is changed. Why do we design the ui in a way that does not take advantage of this? If the Apply/Ok button applies the change and saves the database, then ctrl+s should trigger this. The same applies to naming buttons, as mentioned above. 

Anyway, the ctrl+s discussion is a bit offtopic, should i create a new issue for this?

**Comment by droidmonkey** [comment]
We already have ctrl enter

**Comment by newhinton** [comment]
okay i did not know that. Which kinda supports my argument, why do we use very unusual keybindings for common tasks? 
Usually ctrl+s is associated with saving whats on the screen, and it does not make sense to not use it for that. 

I understand that you make a difference between the database and the entry, but it should not be "visible" to the user unless necessary. And i dont see the necessity for a difference here, since the autosave-default removes the only difference between entry&database

Edit: Also Ctrl+Enter saves&exits, which is a different action from just saving

**Comment by droidmonkey** [comment]
Ctrl+enter is a very common shortcut that means submit form, accept data, etc. I am not going to merge code that double encumbers ctrl+s so please don't argue that point.

**Comment by 7usr7local** [comment]
I'd just like to say that I am completely fine with the UI as it is, using KeepassXC 2.6.6 as shipped with Linux Mint. I am afraid that often trying to make things better make it worse, especially regarding the usability studies droidmonkey mentioned.
If any, i would not remove "OK" but "Apply" as I consider "Apply" the rare case and "OK" and "Cancel" very common options in UI dialogs.
But what I just wrote still holds: Don't risk to make things worse while they're good for most people.
Just my two pence,
Peter (KeepassXC User)

**Comment by foss-** [comment]
Interesting discussion. I found the current implementation a bit confusing. Like what 1Password does as that is a very reduced and approachable solution:

- when an entry is in edit mode they just show `Cancel` and `Save` which seems sufficient
- they allow to enter edit mode with keyboard shortcut `cmd E` (identical with KeePassXC) but then they also allow to save the edited entry with the same short cut `cmd E`

Both solutions they came up with seem smart, easy to remember and avoid confusion on side of users.
